src/sentry/snuba/metrics/span_attribute_extraction.py

```diff
@@ -126,26 +126,86 @@ def _flatten_query_tokens(conditions: Sequence[QueryToken]) -> list[SearchFilter
     return query_tokens
 
 
-def _parse_conditions(conditions: Sequence[str] | None) -> Sequence[QueryToken]:
-    if not conditions:
-        return []
+# Condition string parsing and transformation
+
+
+def _extend_parsed_condtions(parsed_search_query: Sequence[QueryToken]) -> Sequence[QueryToken]:
+    return _visit_numeric_tokens(parsed_search_query)
+
+
+def _visit_numeric_tokens(parsed_search_query: Sequence[QueryToken]) -> list[QueryToken]:
+    """
+    Visits each token in the parsed search query and converts numeric tokens into paren expressions.
+    """
+    query_tokens: list[QueryToken] = []
+
+    for token in parsed_search_query:
+        if isinstance(token, SearchFilter):
+            query_tokens.append(_extend_numeric_token(token))
+        elif isinstance(token, ParenExpression):
+            query_tokens = query_tokens + _visit_numeric_tokens(token.children)
+        else:
+            query_tokens.append(token)
+
+    return query_tokens
+
+
+def _extend_numeric_token(token: SearchFilter) -> ParenExpression | SearchFilter:
+    """
+    Since all search filter values are parsed as strings by default, we need to make sure that
+    numeric values are treated as such when constructing the rule condition. This function
+    expands the original token into a paren expression if the value is a numeric string.
+
+    Example:
+    `key:'123'` -> `key:'123' OR key:123`
+    `key:['123', '456']` -> `key:['123', '456'] OR key:[123, 456]`
+
+    """
+    if token.operator == "=" or token.operator == "!=":
+        if not str(token.value.value).isdigit():
+            return token
+
+        numeric_value_token = SearchFilter(
+            key=token.key, operator=token.operator, value=SearchValue(int(token.value.value))
+        )
+
+    elif token.is_in_filter:
+        str_values = [str(value) for value in token.value.value]
+        if not all(value.isdigit() for value in str_values):
+            return token
+
+        numeric_values = [int(value) for value in str_values]
+        numeric_value_token = SearchFilter(
+            key=token.key, operator=token.operator, value=SearchValue(numeric_values)
+        )
+
+    if not numeric_value_token:
+        return token
```

Comment:
What does this check do? 
Unless I am missing on something, if we enter in one of the if/elif branches, this check will always be true - `numeric_value_token` will be a reference to a `SearchFilter` object.

If non of the conditions in if/elif are true (e.g. operator somehow ends being  '>' or smth similar), then an exception will be raised because we have tried to access variable that wasn't previously defined